test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -84,22 +85,25 @@ public static void main(String[] args) throws IOException {
                             String message = new String(buf.array());
                             System.out.format("received %s at %s from %s%n", message, dc.getLocalAddress(), sa);
                             //If any received data contains the message from sendDatagram then return true
-                            if (message.contains("Greetings!")) {
+                            if (message.contains("Greetings from SelectWhenRefused!")) {
                                 throw new RuntimeException("Unexpected datagram received");
                             }
-                            //unexpected datagram received from elsewhere, retry
-                            continue;
+                            // BindException will be thrown if another service is using
+                            // our expected refuser port, cannot run just exit.
+                            DatagramChannel.open().bind(refuser).close();
+                            throw new RuntimeException("PortUnreachableException not raised");
                         } catch (PortUnreachableException pue) {
                             // expected
                         }
```

Comment:
This should be:
```
                            throw new RuntimeException("PortUnreachableException not raised");
                        } catch (PortUnreachableException pue) {
                            // expected
                        }
                    }
                    break;
```
If we get `PortUnreachableException` we have what we expected and we can exit the loop. Same if n == 0;